readthedocs/core/mixins.py

```diff
@@ -17,3 +18,47 @@ def get_context_data(self, **kwargs):
 class PrivateViewMixin(LoginRequiredMixin):
 
     pass
+
+
+class UpdateChangeReasonForm:
+
+    """Set the change_reason on the model changed through this form."""
+
+    change_reason = 'Changed from: form'
+
+    def save(self, commit=True):
+        obj = super().save(commit=commit)
+        update_change_reason(obj, self.change_reason)
```

Comment:
It seems like we want to put exception handling around all of these `update_change_reason` -- since it sounds like you hit multiple places where it raises exceptions. We don't want the site to break because of this, so we can skip it if needed.

Reply:
Our tests should catch those, if we add this class to a model/view that doesn't support historical records, that code is wrong (same if save doesn't return an object).